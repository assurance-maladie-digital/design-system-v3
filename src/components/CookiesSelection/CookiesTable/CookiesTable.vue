<script setup lang="ts">
	import type { Cookie } from '../types.ts'
	import { headers } from './headers'

	defineProps<{ items: Cookie[] }>()
</script>

<template>
	<VTable class="vd-cookie-table">
		<thead>
			<tr>
				<th
					v-for="(item, index) in headers"
					:key="index"
					:style="{ width: item.width }"
					class="text-left"
				>
					{{ item.label }}
				</th>
			</tr>
		</thead>

		<tbody>
			<tr
				v-for="cookie in items"
				:key="cookie.name"
			>
				<td>{{ cookie.name }}</td>
				<td>
					<slot
						:name="'cookie-description-' + cookie.name"
						:cookie
					>
						{{ cookie.description }}
					</slot>
				</td>
				<td>{{ cookie.conservation }}</td>
			</tr>
		</tbody>
	</VTable>
</template>

<style lang="scss" scoped>
.vd-cookie-table :deep(table) {
	table-layout: fixed;
}
</style>
