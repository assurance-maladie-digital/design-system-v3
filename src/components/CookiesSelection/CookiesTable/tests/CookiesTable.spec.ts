import { describe, it, expect } from 'vitest'
import { shallowMount } from '@vue/test-utils'
import { vuetify } from '@tests/unit/setup'
import CookiesTable from '../CookiesTable.vue'

describe('CookiesTable', () => {
	it('renders correctly', () => {
		const wrapper = shallowMount(CookiesTable, {
			global: {
				plugins: [vuetify],
			},
			propsData: {
				items: [
					{
						name: 'session',
						description: 'Sauvegarde la session pour rester connecté.',
						conservation: '20 heures',
					},
					{
						name: 'cookie_policy',
						description: 'Sauvegarde les préférences de cookies.',
						conservation: '1 an',
					},
				],
			},
		})

		expect(wrapper.html()).toMatchSnapshot()
	})
})
