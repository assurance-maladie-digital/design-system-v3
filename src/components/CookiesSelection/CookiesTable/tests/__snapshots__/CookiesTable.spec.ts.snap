// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CookiesTable > renders correctly 1`] = `"<v-table-stub data-v-d0e8c396="" class="vd-cookie-table bg-transparent" fixedheader="false" fixedfooter="false" hover="false" density="default" tag="div"></v-table-stub>"`;
